streamalert_cli/terraform/helpers.py

```diff
@@ -24,3 +29,33 @@ def terraform_check():
                        'your $PATH:\n'
                        '\t$ export PATH=$PATH:/usr/local/terraform/bin')
     return run_command(['terraform', 'version'], error_message=prereqs_message, quiet=True)
+
+def terraform_state_lock(action, region, table):
+    ddb_client = boto3.client('dynamodb', region_name=region)
+    ddb_tables = ddb_client.list_tables()
+    if action == 'create':
+        if table not in ddb_tables['TableNames']:
+            LOGGER.info('Creating terraform state locking DynamoDB table')
+            ddb_client.create_table(
+                AttributeDefinitions=[{
+                    'AttributeName': 'LockID',
+                    'AttributeType': 'S'
+                }],
+                TableName=table,
+                KeySchema=[{
+                    'AttributeName': 'LockID',
+                    'KeyType': 'HASH'
+                }],
+                BillingMode='PAY_PER_REQUEST'
+            )
+            wait = True
+            while wait:
+                desc_resp = ddb_client.describe_table(TableName=table)
+                if desc_resp['Table']['TableStatus'] == 'ACTIVE':
+                    wait = False
+                else:
+                    time.sleep(1)
+    elif action == 'destroy':
+        if table in ddb_tables['TableNames']:
+            LOGGER.info('Destroying terraform state locking DynamoDB table')
+            ddb_client.delete_table(TableName=table)
```

Comment:
you should also use the [`table_not_exsits`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Waiter.TableNotExists) waiter when deleting the table

Reply:
I thought about doing this. I decided not to because technically it doesn't need to wait for the table to be destroyed since this is one of the last steps and is not required to continue execution. Plus I didn't like the idea of it hanging if the table takes a while to delete. However now that I think about it again, terraform does this and this potentially makes the user experience better so I will add it.

Reply:
my reasoning is that the user likely would not want a rogue/orphaned table hanging around without their knowledge. if the deletion fails and the waiter doesn't succeed, at least an exception will be raise so they know to go an remove manually